pkg/apis/cr/v1alpha1/types.go

```diff
@@ -225,77 +254,112 @@ type BlueprintPhase struct {
 type BlueprintList struct {
 	metav1.TypeMeta `json:",inline"`
 	metav1.ListMeta `json:"metadata"`
-	Items           []*Blueprint `json:"items"`
+	// Items is the list of Blueprints.
+	Items []*Blueprint `json:"items"`
 }
 
 // +genclient
 // +genclient:noStatus
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
-// Profile
+// Profile captures information about a location for data operation artifacts and
+// corresponding credentials that will be made available to a Blueprint.
 type Profile struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata"`
-	Location          Location   `json:"location"`
-	Credential        Credential `json:"credential"`
-	SkipSSLVerify     bool       `json:"skipSSLVerify"`
+	// Specifies the location that the Blueprint can use.
+	Location Location `json:"location"`
+	// Represents the credentials associated with the Location.
+	Credential Credential `json:"credential"`
+	// SkipSSLVerify is boolean and specifies whether skipping SkipSSLVerify verification
+	// is allowed when operating with the Location.
+	// If omitted from a CR definition it default to false
+	SkipSSLVerify bool `json:"skipSSLVerify"`
 }
 
 // LocationType
 type LocationType string
 
 const (
-	LocationTypeGCS         LocationType = "gcs"
+	// LocationType is selected as GCS.
+	LocationTypeGCS LocationType = "gcs"
+	// LocationType is selected as S3 Complaint.
 	LocationTypeS3Compliant LocationType = "s3Compliant"
-	LocationTypeAzure       LocationType = "azure"
-	LocationTypeKopia       LocationType = "kopia"
+	// LocationType is selected as Azure.
+	LocationTypeAzure LocationType = "azure"
+	// LocationType is selected as Kopia.
+	LocationTypeKopia LocationType = "kopia"
 )
 
 // Location
 type Location struct {
-	Type     LocationType `json:"type"`
-	Bucket   string       `json:"bucket"`
-	Endpoint string       `json:"endpoint"`
-	Prefix   string       `json:"prefix"`
-	Region   string       `json:"region"`
+	// Type of the Location being used.
+	Type LocationType `json:"type"`
+	// Represents the bucket being used for the current Location.
+	Bucket string `json:"bucket"`
+	// Represents the endpoints being used by the current Location.
+	Endpoint string `json:"endpoint"`
+	// Prefix is the string used in the beginning of the Location.
+	Prefix string `json:"prefix"`
+	// Represents the region of the current location.
+	Region string `json:"region"`
```

Comment:
This comments are incorrect I think. Let us know the fields that you are unsure about and we can give the details about those fields.